### PR TITLE
Wagtail user experience enhancements to replace raw HTML usage

### PR DESCRIPTION
**Pull request by csebianlander**
This PR compiles a set of changes that address feature needs in Wagtail that were previously being solved using unescaped HTML in plaintext fields. These include:
1. Adding boldface formatting to `Hyperlink` blocks
2. Adding a manual line break in the headings of `Hero` blocks
3. Adding icons and altering padding to `Expandable` block labels

---

## How to test this PR

1. localhost
2. Attempt to create or modify any of the above listed blocks.

## Screenshots

Screenshot 2024-03-05 at 2 13 12 PM
Screenshot 2024-03-05 at 2 13 48 PM
Screenshot 2024-03-05 at 2 15 12 PM

## Checklist

- [ ] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18 (the last Edge prior to it switching to Chromium)
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

File: cfgov/v1/jinja2/v1/includes/molecules/info-unit-2.html
Comment (chosak) [inline cfgov/v1/jinja2/v1/includes/molecules/info-unit-2.html:128]: All of these

`{% if link.make_link_boldface == True %}`

can be simplified to

`{% if link.make_link_boldface %}`

for brevity's sake.

Comment (csebianlander) [reply cfgov/v1/jinja2/v1/includes/molecules/info-unit-2.html]: Resolved in https://github.com/cfpb/consumerfinance.gov/pull/8191/commits/46dfb5ded5a17778287e9baa644e29e4f278e9a6

File: cfgov/v1/jinja2/v1/includes/organisms/expandable.html
Comment (willbarton) [inline cfgov/v1/jinja2/v1/includes/organisms/expandable.html:45]: I might tag @anselmbradford in here to see if there might be another way we want to do the added padding than a style inline.

File: cfgov/v1/jinja2/v1/includes/organisms/expandable.html
Comment (willbarton) [inline cfgov/v1/jinja2/v1/includes/organisms/expandable.html:49]: @anselmbradford here too.

File: cfgov/v1/atomic_elements/molecules.py
Comment (willbarton) [inline cfgov/v1/atomic_elements/molecules.py:92]: While I do see some other help text in this module uses it to do some HTML formatting with the help text, [`mark_safe`](https://docs.djangoproject.com/en/5.0/ref/utils/#django.utils.safestring.mark_safe) looks unnecessary here.

```suggestion
        help_text=(
            "Use if the heading needs to break to a second line"
            "at a specific point in the text."
        ),
```

File: cfgov/v1/jinja2/v1/includes/molecules/info-unit-2.html
Comment (willbarton) [inline cfgov/v1/jinja2/v1/includes/molecules/info-unit-2.html:130]: Pakled saying "We are strong now."

File: cfgov/v1/tests/atomic_elements/organisms/test_organisms.py
Comment (willbarton) [inline cfgov/v1/tests/atomic_elements/organisms/test_organisms.py:305]: Unrelated to your changes, this test and the others like it in this file bother me. I'm not certain why we're testing that this renders with specific whitespace.

File: cfgov/v1/atomic_elements/organisms.py
Comment (anselmbradford) [inline cfgov/v1/atomic_elements/organisms.py:578]: I would probably follow the `is_` style of the other boolean values above here. There's a boolean naming convention to name the variable name like a question with either a prefixed `is_` or a `has_`, which can be answered with "true" or "false" (see examples in https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig).

File: cfgov/v1/jinja2/v1/includes/molecules/hero.html
Comment (anselmbradford) [inline cfgov/v1/jinja2/v1/includes/molecules/hero.html:71]: This works, but it could be shortened to `<br>` if preferred (since we're not in strict XHTML).

File: cfgov/v1/atomic_elements/atoms.py
Comment (anselmbradford) [inline cfgov/v1/atomic_elements/atoms.py:46]: Similar to the other boolean naming comment, you could consider naming this `is_link_boldface`, which is both shorter and more in line with convention.

File: cfgov/v1/atomic_elements/molecules.py
Comment (anselmbradford) [inline cfgov/v1/atomic_elements/molecules.py:87]: This may be too pedantic to matter, but this isn't a heading with a line break, right? It's just a continuation of the heading and then the line break gets added adjacent to it in the template. The only problem with this might be if in the future the `<br />` in the template were to be removed for whatever reason and then we have this named `heading_with_line_break` when it doesn't actually contain a line break. @willbarton would it make any sense to be adding the `<br />` to the text entered here before it ends up in the template? Alternatively I think it would be more accurate as something like `heading_continued` or something, which better conveyed that this is just a continuation of the heading text (which likely would be separated by a line break, but we'll leave that up to the template where it's used).

Comment (willbarton) [reply cfgov/v1/atomic_elements/molecules.py:87]: >  @willbarton would it make any sense to be adding the <br /> to the text entered here before it ends up in the template?

That feels anti-patternish to me, since adding HTML around values is what the templates are for?

File: cfgov/v1/jinja2/v1/includes/organisms/expandable.html
Comment (anselmbradford) [reply cfgov/v1/jinja2/v1/includes/organisms/expandable.html:45]: Yeah this should be added as an example in https://cfpb.github.io/design-system/components/expandables and the padding sorted out there. What is the use case for this again? 

We have an existing `o-expandable__padded` modifier, which should probably be merged into the default expandable (see https://github.com/cfpb/design-system/blob/abd4f9c889f3b8bf2de1a957729e54d88345b8a2/packages/cfpb-expandables/src/expandable.less#L87). That modifier could then be repurposed for this use case.

File: cfgov/v1/jinja2/v1/includes/organisms/expandable.html
Comment (anselmbradford) [reply cfgov/v1/jinja2/v1/includes/organisms/expandable.html:49]: Yes this should be added via a modifier on expandables (see other comment).

File: cfgov/v1/atomic_elements/molecules.py
Comment (anselmbradford) [reply cfgov/v1/atomic_elements/molecules.py:87]: > That feels anti-patternish to me, since adding HTML around values is what the templates are for?

Yeah, but we do add markup in the middleware in some cases, right? For example, with links that get the external link icons added? I wasn't sure if this perhaps fit into a similar convention. It does seem better to just have the markup in a template though, but then again the name `heading_with_line_break` would not be accurate, because it doesn't actually contain the line break.

File: cfgov/v1/tests/atomic_elements/organisms/test_organisms.py
Comment (csebianlander) [reply cfgov/v1/tests/atomic_elements/organisms/test_organisms.py:305]: Yeah it was a huge pain to get that test to resolve accurately due to the arbitrary-ish whitespace. It works for now but is probably worth revisiting?

Comment (willbarton) [reply cfgov/v1/tests/atomic_elements/organisms/test_organisms.py:305]: I think if we revisit, we should revisit that entire file. This isn't the only test like that. So, no action needed for this PR.

File: cfgov/v1/jinja2/v1/includes/molecules/hero.html
Comment (csebianlander) [reply cfgov/v1/jinja2/v1/includes/molecules/hero.html]: changed in https://github.com/cfpb/consumerfinance.gov/pull/8191/commits/374073de5799829c008bc8e4281eddfd4d57220a

File: cfgov/v1/atomic_elements/atoms.py
Comment (csebianlander) [reply cfgov/v1/atomic_elements/atoms.py]: changed in https://github.com/cfpb/consumerfinance.gov/pull/8191/commits/374073de5799829c008bc8e4281eddfd4d57220a

File: cfgov/v1/atomic_elements/organisms.py
Comment (csebianlander) [reply cfgov/v1/atomic_elements/organisms.py]: changed in https://github.com/cfpb/consumerfinance.gov/pull/8191/commits/374073de5799829c008bc8e4281eddfd4d57220a

File: cfgov/v1/atomic_elements/molecules.py
Comment (csebianlander) [reply cfgov/v1/atomic_elements/molecules.py]: changed in https://github.com/cfpb/consumerfinance.gov/pull/8191/commits/374073de5799829c008bc8e4281eddfd4d57220a

File: cfgov/v1/jinja2/v1/includes/organisms/expandable.html
Comment (csebianlander) [reply cfgov/v1/jinja2/v1/includes/organisms/expandable.html]: Merging and utilizing the existing modifier is OK by me but I'm not 100% sure how to do it properly.

Comment (csebianlander) [reply cfgov/v1/jinja2/v1/includes/organisms/expandable.html]: https://github.com/cfpb/consumerfinance.gov/pull/8191/commits/85fcc8622e83222210f131ce0adbb771c5860e2a utilizes the o-expandable__padded modifier per https://github.com/cfpb/design-system/pull/1924 (so we should wait until that ends up in the codebase proper before merging this)

**Comment by anselmbradford** [comment]
https://github.com/cfpb/consumerfinance.gov/pull/8194 is in and should take care of your expandable needs

File: cfgov/v1/jinja2/v1/includes/molecules/hero.html
Comment (willbarton) [inline cfgov/v1/jinja2/v1/includes/molecules/hero.html:72]: There's an errant `p` here it looks like

```suggestion
                {% endif %}
```

File: cfgov/v1/jinja2/v1/includes/organisms/featured-content.html
Comment (willbarton) [inline cfgov/v1/jinja2/v1/includes/organisms/featured-content.html:51]:
```suggestion
                       {% if link.is_link_boldface == True%}<strong>{% endif %}
                       {{- link.text | default( link.url ) -}}
                       {% if link.is_link_boldface == True%}</strong>{% endif %}
```

**Review by willbarton — COMMENTED** [review]
Just a couple more quick things I noticed.

**Review by willbarton — APPROVED** [review]

**Pull request merged**
